### ISSUE TITLE
Implement function to terminate training when the loss is NaN

### ISSUE THREAD

**Issue by cengh** [fchollet/keras#4848]
It would be useful to have training stopped automatically when the loss is NaN at the end of a batch. This functionality would help avoid wasting computational resources when jobs are submitted to a queue.
